torch/_higher_order_ops/invoke.py

```diff
@@ -0,0 +1,66 @@
+import functools
+
+import torch
+import torch.utils._pytree as pytree
+from torch._C import _ExcludeDispatchKeyGuard, DispatchKey, DispatchKeySet
+from torch._higher_order_ops.utils import autograd_not_implemented
+
+from torch._ops import HigherOrderOperator
+from torch._subclasses.fake_tensor import FakeTensorMode
+from torch.fx.experimental.proxy_tensor import ProxyTorchDispatchMode, track_tensor_tree
+from torch.utils._python_dispatch import _get_current_dispatch_mode
+
+
+def invoke(fn, grad):
+    return invoke_op(fn, grad)
+
+
+invoke_op = HigherOrderOperator("invoke")
+
+
+@invoke_op.py_impl(ProxyTorchDispatchMode)
```

Comment:
Pretty cool though :) Is this is our way of (eventually) hiding backward hook from AOTAutograd so that we don't effectively allow_in_graph them, and instead let dynamo inspect them during compiled backwards?